### PR TITLE
[9.0][add] mail_log_messages_to_process

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Mail log messages to process
============================

This module allows to log into the server the basic information of emails
that have been fetched from the mail server, before they start to be processed.

This allows for a better analysis of situations where emails are found to be
missing in Odoo. Can help to better resolve questions of the type 'Was it that
the email was never fetched, or a problem found when it was processed by
Odoo?'.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
cc @lreficent @mreficent @aheficent

**Comment by MiquelRForgeFlow** [comment]
@Yajo, the `needs fixing` tag can be removed  ☺️

File: mail_log_messages_to_process/models/mail_thread.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    _inherit = 'mail.thread'
+
+    @api.model
+    def message_parse_basic_data(self, message):
+        """Parses a string or email.message.Message representing an RFC-2822
+        email, and returns a generic dict holding the message details.
+
+            :param message: the message to parse
+            :rtype: dict
+            :return: A dict with the following structure, where each field
```
Comment (yajo) [inline mail_log_messages_to_process/models/mail_thread.py]: Proper indentation please.

**Review by yajo — APPROVED** [review]

File: mail_log_messages_to_process/models/mail_thread.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        msg_dict['message_id'] = message_id
+        if message.get('Subject'):
+            msg_dict['subject'] = decode(message.get('Subject'))
+
+        # Envelope fields not stored in mail.message but made available
+        # for message_new()
+        msg_dict['from'] = decode(message.get('from'))
+        msg_dict['to'] = decode(message.get('to'))
+        msg_dict['cc'] = decode(message.get('cc'))
+        msg_dict['email_from'] = decode(message.get('from'))
```
Comment (serpentcs-dev1) [inline mail_log_messages_to_process/models/mail_thread.py:56]: `msg_dict['email_from'] = decode(message.get('from'))`
`msg_dict['from'] = decode(message.get('from'))`

any different usage?

**Review by serpentcs-dev1 — APPROVED** [review]
LGTM :+1:

**Pull request merged**